### ISSUE TITLE
Auto net device selection

### ISSUE THREAD

**Issue by ibrokemypie** [greshake/i3status-rust#399]
I switch between ethernet, usb tethering and wifi quite often, is there any chance you could add an option for the net block to automatically select whatever the current default interface is? Or show total across all interfaces?

**Comment by atheriel** [comment]
I believe the NetworkManager block might be a better fit for this. IIUC, it is actually the one in charge of which interface is in use.

**Comment by ibrokemypie** [comment]
The networkmanager block doesnt seem to be documented, how can I get it to show speeds and ip?

**Comment by nicholasfagan** [comment]
A solution to see your multiple interfaces might be to just put multiple `net` blocks, one for each interface and each `net` block will shrink if it is not connected. Although if you had all interfaces active at the same time it would grow rather large.
Currently the `NetworkManager` block only shows the state of an interface, not any stats (i.e. connected, disconnected). 
It might be possible to get the current default interface from parsing the output of `ip route show`, in which case we could add an `auto` option to the `net` block and have it select which is default, but I'll have to read in to how reliable and accurate this is.

**Comment by bohrshaw** [comment]
> It might be possible to get the current default interface from parsing the output of `ip route show`, in which case we could add an `auto` option to the `net` block and have it select which is default, but I'll have to read in to how reliable and accurate this is.

This is good enough for choosing a default interface.

What I think is essential is to dynamically supply the interface names. If the net block can accept a user script to generate the names, that's more flexible.

**Comment by Ploppz** [comment]
Fyi, i3status seems to already have a system wherein, if you define several internet widgets (e.g. wlp1s0, enp4s0, enp2s0), it will only display the info when some device is connected.

What I need is this functionality. Personally I only need to see whether or not I'm connected, and if it's wifi, the SSID.

**Comment by bohrshaw** [comment]
> if you define several internet widgets (e.g. wlp1s0, enp4s0, enp2s0), it will only display the info when some device is connected.

Missing interfaces are shown as "disconnected" instead of "hidden"...

**Comment by Ploppz** [comment]
![ss](https://i.imgur.com/RL0cU7R.png)

My wifi is connected, and my two ethernet interfaces is not, showing two imo useless blocks. I would personally like if any disconnected devices would be hidden. Maybe through configuration, e.g. `hide_disconnected` or something?

**Comment by vvrein** [comment]
`ip link show` might be a solution, but it shows all interfaces, and in case of docker or kvm - interfaces will be pain
```
20:57:56 vrein@tokk:~| ip link show
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: enp6s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP mode DEFAULT group default qlen 1000
    link/ether e0:d5:5e:24:19:2b brd ff:ff:ff:ff:ff:ff
33: vpnwireguarg0: <POINTOPOINT,NOARP,UP,LOWER_UP> mtu 1420 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
    link/none
34: virbr0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN mode DEFAULT group default qlen 1000
    link/ether a6:06:42:40:b7:74 brd ff:ff:ff:ff:ff:ff
35: virbr0-nic: <BROADCAST,MULTICAST> mtu 1500 qdisc fq_codel master virbr0 state DOWN mode DEFAULT group default qlen 1000
    link/ether 52:54:00:c7:6f:d8 brd ff:ff:ff:ff:ff:ff
```
Anothet decision might be `ip route show default`
```
21:01:54 vrein@tokk:~| ip route show  default
default via 192.168.2.1 dev enp6s0 proto dhcp src 192.168.2.3 metric 202
21:01:58 vrein@tokk:~| ip route show  default  | awk '{print $5}'
enp6s0
```

**Comment by nicholasfagan** [comment]
@vvrein I agree. I tested this out a while ago ([here](https://github.com/greshake/i3status-rust/compare/master...nicholasfagan:feature/auto-select-net-device)) and it seemed to work well.

If that seems like a good enough solution to everybody I can make a PR, although I think that @Ploppz suggestion makes more sense to me.

**Comment by wngr** [comment]
@nicholasfagan would be great if you could PR your code. Looks like a pretty reasonable thing to do.

**Comment by ammgws** [comment]
Closed via #626

**Comment by OddBloke** [comment]
Thanks, excited to use this!
